### PR TITLE
Nondestructive logs tests

### PR DESCRIPTION
**Pull request by marusak**

### CODE REVIEW

File: test/verify/check-journal [hunk 1]
```diff
@@ -468,9 +476,13 @@ ExecStart=/bin/sh -c 'for s in $(seq 10); do echo SLOW; sleep 0.1; done; sleep 1
 
     @skipImage("ABRT not available", "debian-stable", "debian-testing", "ubuntu-stable",
                "ubuntu-2004", "fedora-coreos", "rhel-8-3", "rhel-8-3-distropkg", "centos-8-stream")
+    @nondestructive
     def testAbrtSegv(self):
         b = self.browser
+        m = self.machine
 
+        m.execute("systemctl restart abrt-journal-core")
+        self.addCleanup(m.execute, "rm -rf /var/spool/abrt/*")
```
Comment (martinpitt) [inline test/verify/check-journal:516]: At some point it makes sense to split out a TestAbrt class, with this as common setUp(). This would also avoid the duplication of the skipImages. This can be a follow-up, of course.

But let's use restore_dir() instead for consistency?

File: test/verify/check-journal [hunk 2]
```diff
@@ -551,12 +572,12 @@ ExecStart=/bin/sh -c 'for s in $(seq 10); do echo SLOW; sleep 0.1; done; sleep 1
 
         m.upload(["verify/files/mock-bugzilla-server.py"], "/tmp/")
         m.spawn("setsid /tmp/mock-bugzilla-server.py", "mock-bugzilla.log")
-        m.execute("echo 'BugzillaURL=http://localhost:8080' >> /etc/libreport/plugins/bugzilla.conf")
+        m.execute("echo 'BugzillaURL=http://localhost:8080' > /etc/libreport/plugins/bugzilla.conf")
```
Comment (martinpitt) [inline test/verify/check-journal:606]: This should rather use restore_file(). The file may already exist, so let's not clobber it entirely.

File: test/verify/check-journal [hunk 3]
```diff
@@ -468,9 +476,13 @@ ExecStart=/bin/sh -c 'for s in $(seq 10); do echo SLOW; sleep 0.1; done; sleep 1
 
     @skipImage("ABRT not available", "debian-stable", "debian-testing", "ubuntu-stable",
                "ubuntu-2004", "fedora-coreos", "rhel-8-3", "rhel-8-3-distropkg", "centos-8-stream")
+    @nondestructive
     def testAbrtSegv(self):
         b = self.browser
+        m = self.machine
 
+        m.execute("systemctl restart abrt-journal-core")
```
Comment (martinpitt) [inline test/verify/check-journal:515]: Why is that necessary? This happens at the start of the test. If the service needs to be started for the test to work, it should use `start`. If it needs to adjust to config changes or /var/spool/ cleanup, please use `try-restart`.

File: test/verify/check-journal (hunk 1 above)
Comment (marusak) [reply test/verify/check-journal:516]: Yeah, I though the same that it could be split up, but as a follow up.

> But let's use restore_dir() instead for consistency?

Sure, I can do that, but there is not much to restore, we really just need to clean up after ourselves.

File: test/verify/check-journal (hunk 3 above)
Comment (marusak) [reply test/verify/check-journal:515]: Because it listens on journal, but we "restore_file" journal and restart "systemd-journald" and it then does not see new messages in journal.

File: test/verify/check-journal (hunk 1 above)
Comment (martinpitt) [reply test/verify/check-journal:516]: OK, thanks - fine to keep then.

File: test/verify/check-journal (hunk 3 above)
Comment (martinpitt) [reply test/verify/check-journal:515]: Makes sense. Can you please add a comment?

**Comment by martinpitt** [comment]
[One journal-related flake](https://logs.cockpit-project.org/logs/pull-14241-20200617-060722-861b0624-rhel-8-3/log.html#296) about an ambiguous selector. May be unrelated.

**Comment by marusak** [comment]
> One journal-related flake about an ambiguous selector. May be unrelated.

yeah, I saw. It is unrelated, but I'll make the test a bit more robust to prevent this.

**Comment by martinpitt** [comment]
One [check-services](https://logs.cockpit-project.org/logs/pull-14241-20200617-080947-7f4d7c57-fedora-32/log.html#287-2) flake, but let's not be too picky here.
